### ISSUE TITLE
make check fails for 7.3 on Debian 8

### ISSUE THREAD

**Issue by hbhotz** [heimdal/heimdal#289]
=============================================
   Heimdal 7.3.0: tests/kdc/test-suite.log
=============================================

# TOTAL: 18
# PASS:  16
# SKIP:  1
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

SKIP: check-hdb-mitdb
=====================

no MIT KDB support

FAIL: check-iprop
=================

Creating database
Test log recovery
Starting kdc
starting master
starting slave
Waiting for ipropd-slave to start, looking logfile messages.log
Have waited 2 seconds
checking slave is up
Add host
Rollover host keys
killing ipropd s + m + kdc
2017-05-16T12:53:20 syncing slave iprop/slave.test.h5l.se@TEST.H5L.SE from version 36 to version 39
2017-05-16T12:53:20 slave iprop/slave.test.h5l.se@TEST.H5L.SE is now up to date (39)
2017-05-16T12:53:20 Got a signal, but no update in log version 39
2017-05-16T12:53:20 Slave iprop/slave.test.h5l.se@TEST.H5L.SE claims to not have version we already sent to it
2017-05-16T12:53:20 syncing slave iprop/slave.test.h5l.se@TEST.H5L.SE from version 37 to version 39
2017-05-16T12:53:20 slave iprop/slave.test.h5l.se@TEST.H5L.SE is now up to date (39)
2017-05-16T12:53:20 Slave iprop/slave.test.h5l.se@TEST.H5L.SE claims to not have version we already sent to it
2017-05-16T12:53:20 syncing slave iprop/slave.test.h5l.se@TEST.H5L.SE from version 38 to version 39
2017-05-16T12:53:20 slave iprop/slave.test.h5l.se@TEST.H5L.SE is now up to date (39)
2017-05-16T12:53:20 lt-ipropd-master terminated
master down at 2017-05-16T12:53:20

**Comment by nicowilliams** [comment]
Can you check the `heimdal-7-1-branch` branch?  Viktor commited a change to fix a race condition in `check-iprop`.  Thanks.

**Comment by hbhotz** [comment]
That does seem to have fixed it.

(Did it also fix it on Solaris/Illumos? It was a problem there around version 1.2/Solaris 10?)

**Comment by hbhotz** [comment]
Look like I didn’t respond. Sorry. I’m pretty sure I did verify the fix of the specific problem. I found some others I had to work around though. 

When Russ Allbery’s patches are applied, it seems to require the use of LMDB instead of the default Berkeley DB on the system in order to avoid a hang condition for certain admin functions.

I don’t suppose you guys are considering enhancing the PW sync plugin interface so it can sync other state, like "locked”, as well?

> On Sep 25, 2017, at 3:23 PM, Nico Williams <notifications@github.com> wrote:
> 
> Closed #289.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 

Personal email.  hbhotz@oxy.edu
